File: java/problem849.java
class Solution {
    public int maxDistToClosest(int[] seats) {
        if(seats.length<2){
            return 0;
        }
        int i=0,j=0;
        boolean tag=false; 
        int max_dis=0;
        while(j<seats.length&&i<seats.length){
            if(seats[i]!=0){
                i++;
                j++;
            }else{
                tag=true;
            }
            if(tag){
                if(seats[j]==0&&j!=seats.length-1){
                    j++;
                }else{
                    tag=false;
                    if(i==0){
                        if((j-i)>max_dis){
                            max_dis=j-i;
                        }
                    }else if(j==seats.length-1&&seats[j]==0){
                        if((j-i+1)>max_dis){
                            max_dis=j-i+1;
                        }
                        break;
                    }else{
                        if((j-i+1)/2>max_dis){
                            max_dis=(j-i+1)/2;
                        }
                    }
                    i=j;
                }
            }
        } 
        return max_dis;     
    }
}

File: python/problem849.py
import math
class Solution:
    def maxDistToClosest(self, seats):
        """
        :type seats: List[int]
        :rtype: int
        """
        num = 0
        firstmax = 0
        for i in range(len(seats)):
            if seats[i] == 0:
                firstmax+=1
            if seats[i] == 1:
                num = i
                break
            
        max_dis = 0
        tempmax = 0
        for i in range(num,len(seats)):
            
            if seats[i] == 0:
                tempmax+=1
            if seats[i] == 1:
                if math.ceil(tempmax/2) > max_dis:
                    max_dis = math.ceil(tempmax/2)
                tempmax=0
        
        return max(max_dis,tempmax,firstmax)
